tokio-console/src/state/mod.rs

```diff
@@ -511,15 +511,20 @@ fn truncate_registry_path(s: String) -> String {
 
     static REGEX: OnceCell<Regex> = OnceCell::new();
     let regex = REGEX.get_or_init(|| {
-        Regex::new(r".*/\.cargo(/registry/src/[^/]*/|/git/checkouts/)")
-            .expect("failed to compile regex")
+        Regex::new(
+            r".*(/|\\)\.cargo(/|\\)(registry(/|\\)src(/|\\)[^/\\]*(/|\\)|git(/|\\)checkouts(/|\\))",
+        )
+        .expect("failed to compile regex")
     });
 
-    return match regex.replace(&s, "<cargo>/") {
+    let s = match regex.replace(&s, "<cargo>/") {
         Cow::Owned(s) => s,
         // String was not modified, return the original.
-        Cow::Borrowed(_) => s.to_string(),
+        Cow::Borrowed(_) => s,
     };
+
+    // This help use the same path separator on all platforms.
+    s.replace('\\', "/")
```

Comment:
I'm not sure this is a good idea, it might cause issues if people have a `/` or `\` in some directory name (although I'm not sure we should worry about that case really).

Also, while `"C:/Users/user/projects/tokio-1.0.1/src/lib.rs"` looks fine to me, maybe it looks weird to a Windows user?

Reply:
> Also, while `"C:/Users/user/projects/tokio-1.0.1/src/lib.rs"` looks fine to me, maybe it looks weird to a Windows user?

The reason I want to use the same path style is that users can use tokio-console to connect to servers from different servers. So maybe using the same path style would help us reduce confusion. But I also agree that it might cause some confusion for Windows users.

Reply:
> I'm not sure this is a good idea, it might cause issues if people have a `/` or `\` in some directory name (although I'm not sure we should worry about that case really).

Yes. This is possible. At least, on macOS, I can create a dir that contains it by command: `mkdir test\\a`.

Reply:
How about:
1. `<cargo>/tokio-1.0.1/src/lib.rs` for macOS and Linux path.
2. `<cargo>\tokio-1.0.1\src\lib.rs` for Windows path.

But I am not sure if there is an easy way to detect if the path string is from the Windows OS.

Reply:
@hds Do you have any thoughts on this? I am not sure which way we should go.

Reply:
> How about:
> 
>     1. `<cargo>/tokio-1.0.1/src/lib.rs` for macOS and Linux path.
> 
>     2. `<cargo>\tokio-1.0.1\src\lib.rs` for Windows path.
> 
> 
> But I am not sure if there is an easy way to detect if the path string is from the Windows OS.

Let's go with this. I would suggest we just go with what separator we see in the path. If it's `/` we leave it as `/`, if it's `\` we leave it `\`. Naively you could probably count the number of each character in the string and only use `\` (which I would consider less likely) if it is stricly higher.

Reply:
I used:
```rust
fn is_windows_path(path: &str) -> bool {
    use once_cell::sync::OnceCell;
    use regex::Regex;

    static REGEX: OnceCell<Regex> = OnceCell::new();
    let regex = REGEX.get_or_init(|| Regex::new(r"^[a-zA-Z]:\\").expect("failed to compile regex"));
    let has_drive_letter = regex.is_match(path);
    let slash_count = path.find('/').iter().count();
    let backslash_count = path.find('\\').iter().count();
    has_drive_letter && backslash_count > slash_count
}
```

It is not perfect, but it seems to work.